translate/translate.go

```diff
@@ -79,42 +85,66 @@ func couldBeValidTranslator(t reflect.Type) bool {
 	if t.Kind() != reflect.Func {
 		return false
 	}
-	if t.NumIn() != 1 || t.NumOut() != 1 {
+	if t.NumIn() != 1 || t.NumOut() != 2 {
 		return false
 	}
-	if util.IsInvalidInConfig(t.In(0).Kind()) || util.IsInvalidInConfig(t.Out(0).Kind()) {
+	if util.IsInvalidInConfig(t.In(0).Kind()) ||
+		util.IsInvalidInConfig(t.Out(0).Kind()) ||
+		t.Out(1) != translationsType {
 		return false
 	}
 	return true
 }
 
+// fieldName returns the name uses when (un)marshalling a field. t should be a reflect.Value of a struct,
+// index is the field index, and tag is the struct tag used when (un)marshalling (e.g. "json" or "yaml")
+func fieldName(t reflect.Value, index int, tag string) string {
```

Comment:
Should this also do an initial assert that `t.Kind() == reflect.Struct`?

Reply:
And do what? We could make it an error but that would only come up due to programming errors and would be propagated all over the place. We could make it panic that's what it already does.

Reply:
That's fair, the base panic message will probably be sufficient.